system-helper/flatpak-system-helper.c

```diff
@@ -247,6 +360,33 @@ handle_deploy (FlatpakSystemHelper   *object,
       return TRUE;
     }
 
+  src_dir = g_path_get_dirname (arg_repo_path);
+  ongoing_pull = take_ongoing_pull_by_dir (src_dir);
+  if (ongoing_pull != NULL)
+    {
+      g_autoptr(GError) local_error = NULL;
+      gint dir_fd = -1;
```

Comment:
glnx_autofd missing here? Otherwise we're leaking dir_fd...

Reply:
Actually, why are you doing this at all? arg_repo_path is an absolute path, no? So instead of passing dir_fd you can pass AT_FDCWD to flatpak_canonicalize_permissions.